nova_vm/src/ecmascript/builtins/array.rs

```diff
@@ -44,6 +44,21 @@ impl Array {
         Self(ArrayIndex::from_u32_index(0))
     }
 
+    /// Creates a new array with the given elements that are converted to values.
+    pub fn from_slice(agent: &mut Agent, elements: &[impl IntoValue]) -> Self {
+        let array = agent.heap.create(ArrayHeapData::default());
```

Comment:
It'd be better to reserve the left upfront:

```suggestion
        let mut heap_data = ArrayHeapData::default();
        heap_data.elements.reserve(
            &mut agent.heap.elements,
            u32::try_from(elements.len()).unwrap(),
        );
        let array = agent.heap.create(heap_data);
```